addons/account/models/account_payment_term.py

```diff
@@ -205,49 +228,69 @@ def unlink(self):
             ).unlink()
         return super(AccountPaymentTerm, self).unlink()
 
+    def _get_last_discount_date(self, date_ref):
+        for pay_term in self:
+            return date_ref + relativedelta(days=(pay_term.discount_days or 0))
+
+    def _get_last_discount_date_formatted(self, date_ref):
+        for pay_term in self:
+            return format_date(self.env, date_ref + relativedelta(days=(pay_term.discount_days or 0)))
```

Comment:
```suggestion
        self.ensure_one()
        return format_date(self.env, date_ref + relativedelta(days=self.discount_days or 0))
```
EDIT:
```suggestion
        self.ensure_one()
        return format_date(self._get_last_discount_date())
```

Reply:
Impossible since we don't have `date_ref`